src/docs/codelabs/extended-flex-widget.md

```diff
@@ -0,0 +1,597 @@
+<iframe src="https://dartpad.dev/experimental/embed-new-flutter.html?id=6582851e85b57180ff5321f814fabb81&amp;theme=dark&amp;split=60" width="100%" height="400px"></iframe>
+
+#### Example: Creating space 
+{:.no_toc}
+{{site.alert.secondary}}
+  The following example contains three `BlueBox` widgets and one `SizedBox` widget that separates the first and second `BlueBox` widgets. The `SizedBox` widget contains a `width` property equal to 50 logical pixels. 
+
+  **1.** Click the **Run** button.
+
+  **2.** Create more space by adding another `SizedBox` widget (25 logical pixels wide) 
+  between the second and third `BlueBox` widgets, and run again.
+{{site.alert.end}}
+{% comment %}
+  Gist: https://gist.github.com/datafoya/19ead147ab5c7668d7d32e1cfed90097
+{% endcomment %}
+<iframe src="https://dartpad.dev/experimental/embed-new-flutter.html?id=19ead147ab5c7668d7d32e1cfed90097&amp;theme=dark&amp;split=60" width="100%" height="400px"></iframe>
+
+## Spacer widget
+
+Similar to the `SizedBox` widget, the `Spacer` widget also can create space between widgets.
+
+{{site.alert.tip}}
+  **What's the difference between SizedBox and Spacer?**
+    Use `Spacer` when you want to create space using a `flex` property.
+    Use `SizedBox` when you want to create space 
+    using a specific number of logical pixels.
+{{site.alert.end}} 
+
+#### Example: Creating more space
+{:.no_toc}
+{{site.alert.secondary}}
+  The following example separates the first two `BlueBox` widgets using
+  a `Spacer` widget with a `flex` value of 1.
+
+  **1.** Click the **Run** button.
+
+  **2.** Add another `Spacer` widget (also with a `flex` value of 1)
+   between the second and third `BlueBox` widgets.  
+{{site.alert.end}}
+{% comment %}
+  Gist: https://gist.github.com/datafoya/bfc367aefde35e02ea5283efdbf58e60
+{% endcomment %}
+<iframe src="https://dartpad.dev/experimental/embed-new-flutter.html?id=bfc367aefde35e02ea5283efdbf58e60&amp;theme=dark&amp;split=60" width="100%" height="400px"></iframe>
+
+## Text widget
+
+The `Text` widget displays text and can be configured
+for different fonts, sizes, and colors. 
+
+#### Example: Aligning text
+{:.no_toc}
+{{site.alert.secondary}}
+  The following example displays "Hey!" three times in one row,
+  but at different font sizes and in different colors.
+  `Row` specifies the `crossAxisAlignment` and `textBaseline` properties.
+
+  As mentioned previously, to align `Text` widgets along their character baselines,
+  set the `textBaseline` property to `TextBaseline.alphabetic`.  
+
+  **1.** Click the **Run** button. 
+
+  **2.** Change `CrossAxisAlignment.center` to `CrossAxisAlignment.baseline`, and run again.
+{{site.alert.end}}
+{% comment %}
+  Gist: https://gist.github.com/datafoya/0ff109090b99ef1873d9fad501b2bc86
+{% endcomment %}
+<iframe src="https://dartpad.dev/experimental/embed-new-flutter.html?id=0ff109090b99ef1873d9fad501b2bc86&amp;theme=dark&amp;split=60" width="100%" height="400px"></iframe>
+
+## Icon widget
```

Comment:
What do the Text & Icon & Image widgets have to do with layout? I guess the Text one is kind of related, but the Icon & Image ones??? Maybe you could delete these examples and just feed them the code later when they need it?